spec/controllers/state_file/archived_intake/mailing_address_validation_controller_spec.rb

```diff
@@ -0,0 +1,113 @@
+require "rails_helper"
+
+RSpec.describe StateFile::ArchivedIntakes::MailingAddressValidationController, type: :controller do
+  let(:intake) { create(:state_file_archived_intake, mailing_state: "NY") }
+  let(:current_request) { create(:state_file_archived_intake_request, email_address:email_address, failed_attempts: 0, state_file_archived_intake: intake) }
+  let(:controller_instance) { described_class.new }
+  let(:email_address) { "test@example.com" }
+  let(:valid_verification_code) { "123456" }
+  let(:invalid_verification_code) { "654321" }
+
+  before do
+    Flipper.enable(:get_your_pdf)
+    allow(controller).to receive(:current_request).and_return(current_request)
+    allow(I18n).to receive(:locale).and_return(:en)
+    session[:code_verified] = true
+    session[:ssn_verified] = true
+  end
+
+  describe "GET #edit" do
+    context "when the request is locked" do
+      before do
+        allow(current_request).to receive(:access_locked?).and_return(true)
+      end
+
+      it "redirects to error page" do
+        get :edit
+
+        expect(response).to redirect_to(state_file_archived_intakes_verification_error_path)
+      end
+    end
+
+    context "when the request is not locked" do
+      before do
+        allow(current_request).to receive(:access_locked?).and_return(false)
+      end
+
+      it "renders the edit template with a new MailingAddressValidationForm" do
+        get :edit
+
+        expect(assigns(:form)).to be_a(StateFile::ArchivedIntakes::MailingAddressValidationForm)
+        expect(response).to render_template(:edit)
+      end
+    end
+
+    it "redirects to the lockout path when the request is locked" do
```

Comment:
what's the difference between this and the test on line 25?

Reply:
nothing oops.